### DISCUSSION TITLE
How to add two columns?

### DISCUSSION THREAD

**Discussion by newtonharry** [vbarrielle/sprs#349] [General]
Hi,

Sorry for the simple question, but I'm having a hard time trying to add two columns in a csc matrix. Here is my current naive approach as I'm having a hard time trying to understand the API:

```rust
let mut acc: CsVec<i64> = CsVec::empty(matrix.rows());

for col_vec in matrix.outer_iterator() {
  col_vec.iter().for_each(|(row_ind, value)| {
      *acc.get_mut(row_ind).unwrap_or(&mut 0) += value;
  });
}
```

**Comment by mulimoen** [comment]
Does this do what you expect?
```rust
let mut acc: CsVec<i64> = CsVec::empty(matrix.rows());

for col_vec in matrix.outer_iterator() {
  acc = acc + col_vec;
}
```
There is no `add_assign` since addition might change the sparse structure of the vector.

**Reply by newtonharry** [reply]
This worked, thanks!
